File: stacked-web/lib/theme/due-date-chip-style.ts
export type DueDateChipStyle = "soft" | "flat" | "plain" | "day" | "outline";

export const DUE_DATE_CHIP_STYLE_KEY = "stacked.dueDateChipStyle";
export const DUE_DATE_CHIP_STYLE_EVENT = "stacked:due-date-chip-style";
export const DEFAULT_DUE_DATE_CHIP_STYLE: DueDateChipStyle = "soft";

export const DUE_DATE_CHIP_STYLES: {
  id: DueDateChipStyle;
  name: string;
  subtitle: string;
}[] = [
  { id: "soft", name: "Suave", subtitle: "Fundo translúcido, borda e calendário (atual)" },
  { id: "flat", name: "Plano", subtitle: "Calendário + texto na cor, sem container" },
  { id: "plain", name: "Texto", subtitle: "Só o texto colorido, sem ícone" },
  { id: "day", name: "Dia", subtitle: "Número do dia real + rótulo ao lado" },
  { id: "outline", name: "Traço", subtitle: "Contorno fino, sem preenchimento" },
];

export function parseDueDateChipStyle(raw: string | null | undefined): DueDateChipStyle {
  if (
    raw === "flat" ||
    raw === "plain" ||
    raw === "day" ||
    raw === "outline" ||
    raw === "soft"
  ) {
    return raw;
  }
  return DEFAULT_DUE_DATE_CHIP_STYLE;
}

export function readDueDateChipStyle(): DueDateChipStyle {
  if (typeof window === "undefined") return DEFAULT_DUE_DATE_CHIP_STYLE;
  return parseDueDateChipStyle(window.localStorage.getItem(DUE_DATE_CHIP_STYLE_KEY));
}

export function writeDueDateChipStyle(style: DueDateChipStyle) {
  if (typeof window === "undefined") return;
  window.localStorage.setItem(DUE_DATE_CHIP_STYLE_KEY, style);
  window.dispatchEvent(new Event(DUE_DATE_CHIP_STYLE_EVENT));
}

export function subscribeDueDateChipStyle(onStoreChange: () => void) {
  if (typeof window === "undefined") return () => {};
  const handler = () => onStoreChange();
  window.addEventListener(DUE_DATE_CHIP_STYLE_EVENT, handler);
  window.addEventListener("storage", handler);
  return () => {
    window.removeEventListener(DUE_DATE_CHIP_STYLE_EVENT, handler);
    window.removeEventListener("storage", handler);
  };
}

File: stacked-ios/Stacked/DesignSystem/DueDateChipStyle.swift
import SwiftUI

/// Visual da data na meta line de tarefas/subtarefas (Aparência).
enum DueDateChipStyle: String, CaseIterable, Identifiable {
  /// Atual: fill + borda + ícone (agora sem dia fixo no glyph).
  case soft
  /// Ícone + texto na cor, sem container.
  case flat
  /// Só o texto colorido (“Hoje”, “17 jul”).
  case plain
  /// Badge com o dia do mês real + rótulo.
  case day
  /// Contorno fino, sem fill.
  case outline

  var id: String { rawValue }

  var displayName: String {
    switch self {
    case .soft: "Suave"
    case .flat: "Plano"
    case .plain: "Texto"
    case .day: "Dia"
    case .outline: "Traço"
    }
  }

  var subtitle: String {
    switch self {
    case .soft: "Fundo translúcido, borda e calendário (atual)"
    case .flat: "Calendário + texto na cor, sem container"
    case .plain: "Só o texto colorido, sem ícone"
    case .day: "Número do dia real + rótulo ao lado"
    case .outline: "Contorno fino, sem preenchimento"
    }
  }
}

enum DueDateChipStyleStorage {
  static let key = "dueDateChipStyle"

  static var defaultRawValue: String { DueDateChipStyle.flat.rawValue }

  static func style(from rawValue: String) -> DueDateChipStyle {
    DueDateChipStyle(rawValue: rawValue) ?? .flat
  }

  static var current: DueDateChipStyle {
    style(from: UserDefaults.standard.string(forKey: key) ?? defaultRawValue)
  }
}
